addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/messages/RFXComLighting2Message.java

```diff
@@ -311,47 +269,48 @@ public void setDeviceId(String deviceId) throws RFXComException {
     }
 
     @Override
-    public void convertFromState(RFXComValueSelector valueSelector, Type type) throws RFXComException {
+    public void convertFromState(String channelId, Type type) throws RFXComException {
 
-        switch (valueSelector) {
-            case COMMAND:
+        switch (channelId) {
+            case CHANNEL_COMMAND:
                 if (type instanceof OnOffType) {
                     if (group) {
                         command = (type == OnOffType.ON ? GROUP_ON : GROUP_OFF);
+
                     } else {
                         command = (type == OnOffType.ON ? Commands.ON : Commands.OFF);
                     }
+
                     dimmingLevel = 0;
                 } else {
-                    throw new RFXComException("Can't convert " + type + " to Command");
+                    throw new RFXComException("Channel " + channelId + " does not accept " + type);
```

Comment:
We could also use early returns in the method `convertFromState`  then outside of the switch we just need a single `throw new RFXComException("Channel " + channelId + " does not accept " + type);`